frontend/packages/pipelines-plugin/src/components/repository/list-page/RepositoryRow.tsx

```diff
@@ -0,0 +1,115 @@
+import { TableRow, TableData, RowFunction } from '@console/internal/components/factory';
+import { Kebab, ResourceKebab, ResourceLink, Timestamp } from '@console/internal/components/utils';
+import { referenceFor, referenceForModel } from '@console/internal/module/k8s';
+import { PipelineRunModel, RepositoryModel } from '../../../models';
+import { pipelineRunFilterReducer } from '../../../utils/pipeline-filter-reducer';
+import { pipelineRunDuration } from '../../../utils/pipeline-utils';
+import LinkedPipelineRunTaskStatus from '../../pipelineruns/status/LinkedPipelineRunTaskStatus';
+import PipelineRunStatus from '../../pipelineruns/status/PipelineRunStatus';
+import { useGetLastPLR } from '../repository-hooks';
+import { RepositoryKind } from '../types';
+import { repositoriesTableColumnClasses } from './RepositoryHeader';
+
+type RepositoryPLRProps = {
+  pipelineRunsName: string[];
+  namespace: string;
+};
+
+export const PLRStatus: React.FC<RepositoryPLRProps> = ({ pipelineRunsName, namespace }) => {
+  const latestRun = useGetLastPLR(pipelineRunsName, namespace);
+  return (
+    <PipelineRunStatus
+      status={pipelineRunFilterReducer(latestRun)}
+      title={pipelineRunFilterReducer(latestRun)}
+      pipelineRun={latestRun}
+    />
+  );
+};
+
+export const PLRRuntime: React.FC<RepositoryPLRProps> = ({ pipelineRunsName, namespace }) => {
+  const latestRun = useGetLastPLR(pipelineRunsName, namespace);
+  return <Timestamp timestamp={latestRun?.status.startTime} />;
+};
+
+export const PLRDuration: React.FC<RepositoryPLRProps> = ({ pipelineRunsName, namespace }) => {
+  const latestRun = useGetLastPLR(pipelineRunsName, namespace);
+  return <>{pipelineRunDuration(latestRun)}</>;
+};
+
+export const PLRLink: React.FC<RepositoryPLRProps> = ({ pipelineRunsName, namespace }) => {
+  const latestRun = useGetLastPLR(pipelineRunsName, namespace);
+  return (
+    <ResourceLink
+      kind={referenceForModel(PipelineRunModel)}
+      name={latestRun?.metadata.name}
+      namespace={namespace}
+    />
+  );
+};
+
+export const PLRTask: React.FC<RepositoryPLRProps> = ({ pipelineRunsName, namespace }) => {
+  const latestRun = useGetLastPLR(pipelineRunsName, namespace);
+  return latestRun && <LinkedPipelineRunTaskStatus pipelineRun={latestRun} />;
+};
+
+const RepositoryRow: RowFunction<RepositoryKind> = ({ obj, index, key, style }) => {
+  const pipelineRunsName = obj.pipelinerun_status?.map((plrStatus) => plrStatus.pipelineRunName);
+  const {
+    metadata: { name, namespace },
+  } = obj;
+  return (
+    <TableRow
+      id={obj.metadata.uid}
+      data-test-id={`${namespace}-${name}`}
+      index={index}
+      trKey={key}
+      style={style}
+    >
+      <TableData className={repositoriesTableColumnClasses[0]}>
+        <ResourceLink kind={referenceForModel(RepositoryModel)} name={name} namespace={namespace} />
+      </TableData>
+      <TableData className={repositoriesTableColumnClasses[1]}>{obj?.spec?.event_type}</TableData>
```

Comment:
Are you worried about `obj` not being defined? Are you worried about `obj.spec` not being defined?

I think there is too much optional chaining here.

Reply:
Updated this.